src/Plutip/Server.purs

```diff
@@ -209,7 +240,33 @@ startPlutipCluster cfg utxoDistribution = do
           <<< (decodeAeson <=< parseJsonStringToAeson)
           <<< _.body
       )
-  either (liftEffect <<< throw <<< show) pure res
+  either (liftEffect <<< throw <<< show) pure res >>=
+    case _ of
+      ClusterStartupFailure _ -> do
+        liftEffect $ throw "Failed to start up cluster"
+      ClusterStartupSuccess response@{ privateKeys } ->
+        case Array.uncons privateKeys of
+          Nothing ->
+            liftEffect $ throw $
+              "Impossible happened: insufficient private keys provided by plutip. Please report as bug."
+          Just { head: PrivateKeyResponse ourKey, tail } ->
+            pure $ PrivatePaymentKey ourKey /\ response { privateKeys = tail }
+
+-- | Calculate the initial utxos needed for `ourKey` to cover
+-- | transaction costs for the given initial distribution
+ourInitialUtxos :: InitialUTxODistribution -> InitialUTxO
+ourInitialUtxos utxoDistribution =
+  let
+    numUtxos = foldr (length >>> add) 0 utxoDistribution
+  in
+    [ -- For each utxo, include an excessive amount of ada to cover
+      -- the transaction fees of including that utxo. Also make sure
+      -- we don't request a 0 ada utxo
+      BigInt.fromInt 1_000_000_000 +
+        (BigInt.fromInt numUtxos * BigInt.fromInt 1_000_000_000)
+    , -- In case a collateral utxo is needed
```

Comment:
It is not, because we don't use scripts with this wallet. Please remove the second UTxO.